### PR TITLE
Feat/swarinfo 353 interloc api to host

### PR DESCRIPTION
**Pull request by xgroleau**

### CODE REVIEW

File: src/pheromones/src/GetNeighborResponseDTO.cpp
Comment (t-constantin) [inline src/pheromones/src/GetNeighborResponseDTO.cpp:31]: if no m_pos, isn't because m_pos is null?
If so, m_pos.has_value() might create a nullptr exception

File: src/pheromones/src/GetNeighborsListResponseDTO.cpp
Comment (t-constantin) [inline src/pheromones/src/GetNeighborsListResponseDTO.cpp:35]: what differs this method from the upper one?

**Review by t-constantin — APPROVED** [review]
LGTM
Just few questions, but I would maybe wait for another approval before merging as I am not that experienced with that code

File: src/pheromones/src/GetNeighborResponseDTO.cpp
Comment (xgroleau) [reply src/pheromones/src/GetNeighborResponseDTO.cpp:31]: It's not a pointer here, it's an std::optional, which implement boo() (what is used in an if conditional). So you can just check with an if it has a value. has_value is a method from std::optional. 

doing 

```cpp
std::optional<int> optionalVar = {}; // empty
if(optionalVar) {
// will not  be called
}
```

is the same as 

```cpp
std::optional<int> optionalVar = {}; // empty
if(optionalVar.has_value()) {
// will not  be called
}
```

just the second one is more verbose for no reason

File: src/pheromones/src/GetNeighborsListResponseDTO.cpp
Comment (xgroleau) [reply src/pheromones/src/GetNeighborsListResponseDTO.cpp:35]: Good catch, not necessary, wil remove. was during dev

Comment (xgroleau) [reply src/pheromones/src/GetNeighborsListResponseDTO.cpp:35]: Good catch, not necessary, will remove. Forgot to remove during cleanup

File: src/pheromones/src/GetNeighborResponseDTO.cpp
Comment (t-constantin) [reply src/pheromones/src/GetNeighborResponseDTO.cpp:31]: nice, good to know!

**Review by cquesnel — APPROVED** [review]

**Pull request merged**
